_docs/dsl.md

```diff
@@ -235,6 +240,18 @@ m.Match(`!!$x`).Suggest(`$x`)
 m.Match(`!!$x`).Suggest(`$x`).Report(`suggested: $x`)
 ```
 
+Be careful when using `Suggest()` with `MatchComment()`. An entire comment node will be rewritten by the rule, so in this case you can't rely on the partial matching. Match an entire comment and use named groups to reconstruct the fixed form of the comment:
+
+```go
+// Bad! If the comment has some other text, it will be removed.
+// "// nolint foo bar" => "//nolint"
+m.MatchComment(`// nolint`).Suggest(`//nolint`)
+
+// Good. No information loss in this case.
+// "// nolint foo bar" => "//nolint foo bar"
+m.MatchComment(`// nolint(?P<rest>)`).Suggest(`//nolint$rest`)
```

Comment:
In practice, that should be able to handle several spaces after `//` and optional `:`

Reply:
I removed this part from the docs.
See PR with actual implementation.